### DISCUSSION TITLE
Sub menu?

### DISCUSSION THREAD

**Discussion by msman0** [poblabs/weewx-belchertown#873] [Q&A]
Hi, how can I add sub menu items (drop down) as shown at these page?
[https://www.woellsdorf-wetter.de/](https://www.woellsdorf-wetter.de/)
Thank you
Regards

**Comment by teffi71** [comment]
@roe-dl would you mind sharing? Please!

**Comment by teffi71** [comment]
@msman0 did you find a solution?

I also looked into this.

This is what I have for adding an additional top-level menu item:

My notes, to manually adding new page to Belchertown skin Menu:

1. skin.conf

- `# Navigation Menu`, add:
nav_'newPage' = 'newPage'
- `# Default page headers`, add:
'newPage'_page_header = " 'newPage header' "
- CheetahGenerator section, add:
```
[CheetahGenerator]
   [[ToDate]]
     [[['newPage']]]
        template = 'newPageFolder'/index.html.tmpl
```

2. header.html.tmpl

Under `<section id="nav_menu">`, add new `<li class="menu-item menu-item-#'>`, 
reordering menu-items in desired order; 
for example, 'newPage' as 2nd menu-item after Home, add:

```
<li class="menu-item menu-item-1">
<a href="$relative_url/'newPageFolder'" itemprop="url">
<span itemprop="name">$obs.label.nav_'newPage'</span></a></li>
```

reordering following menu-items: 2,3,4,5

3. page-header.inc

Add after `#if $page == "graphs" <h1>...</h1>`:

```
#else if $page == "'newPage'"
   <h1>$obs.label.'newPage'_page_header</h1>
```

4.  New Folder

Create folder 'newPageFolder'
Copy 'about.inc.example' into 'newPageFolder', and edit accordingly

**Comment by roe-dl** [comment]
Sorry, I was ill for a long time. I did not see discussion.

The logic of sub-menues is already included in the web framework used for the Belchertown skin. So you can simply add items. The menu is a HTML list. It starts with `<ul>` and ends with `</ul>`. In between are items surrounded by `<li>...</li>`. If you want a sub-menu, you put it between the `<li>...</li>` of the respective main menu item. There you start again with `<ul>` and end with `</ul>` and put the sub-menu items in between.

All menu items get the class `menu-item` and another class `menu-item-<number>` where `<number>` is an arbitrary number. Every number should appear only once, but they do not need to be in order. 

The sub-menu `<ul>` gets the classes `menu menu-menu sub-menu responsive-menu genesis-nav-menu`. So the whole statement is `<ul class="menu menu-menu sub-menu responsive-menu genesis-nav-menu">`.

In `custom.css` you may add some styling, especially to get a solid background of the opened sub-menu. I did it that way:
```Python
.genesis-nav-menu a {
        color: #333;
        display: block;
        padding: 30px 24px 10px;
}

.genesis-nav-menu .sub-menu a {
  padding: 15px;
}

.genesis-nav-menu .sub-menu {
  /* offset-x offset-y blur-radius spread-radius color */
  box-shadow: 0px 0px 3px 1px rgba(200, 200, 200, 0.2);
}

.genesis-nav-menu .sub-menu .sub-menu {
  margin: -46px 0 0 199px;
  box-shadow: 0px 0px 3px 1px rgba(200, 200, 200, 0.2);
}

.dark .sub-menu {
  background-color:#333;
}

.light .sub-menu {
  background-color:#FFF;
}
```
